src/main/java/org/akaza/openclinica/controller/SdvApiController.java

```diff
@@ -47,29 +50,37 @@
-        studyBuildService.setRequestSchemaByStudyOrParentStudy(studyOID);
-            UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
-        boolean changedAfterSdvOnlyFilterFlag=true;
-        int ordinalValue ;
-
-        if(changedAfterSdvOnlyFilter.equals("n"))
-            changedAfterSdvOnlyFilterFlag = false;
-        SdvDTO responseDTO = null;
         try {
-            if (ordinal == null)
-                ordinalValue = 1;
-            else if(ordinal.matches("^[0-9]*$") && Integer.parseInt(ordinal) > 0)
-                ordinalValue = Integer.parseInt(ordinal);
-            else
-                throw new OpenClinicaSystemException( ErrorConstants.ERR_EVENT_ORDINAL_IS_INCORRECT);
-            validateService.validateForSdvItemForm(studyOID, studyEventOID, studySubjectLabel, formOID, userAccountBean,ordinalValue);
-            responseDTO = sdvUtil.getFormDetailsForSDV(studyOID, formOID, studyEventOID, studySubjectLabel, ordinalValue, changedAfterSdvOnlyFilterFlag);
+            studyBuildService.setRequestSchemaByStudyOrParentStudy(studyOID);
+            UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
+            boolean changedAfterSdvOnlyFilterFlag=true;
+            int ordinalValue ;
+    
+            if(changedAfterSdvOnlyFilter.equals("n"))
+                changedAfterSdvOnlyFilterFlag = false;
+            SdvDTO responseDTO = null;
+            try {
+                if (ordinal == null)
+                    ordinalValue = 1;
+                else if(ordinal.matches("^[0-9]*$") && Integer.parseInt(ordinal) > 0)
+                    ordinalValue = Integer.parseInt(ordinal);
+                else
+                    throw new OpenClinicaSystemException( ErrorConstants.ERR_EVENT_ORDINAL_IS_INCORRECT);
+                validateService.validateForSdvItemForm(studyOID, studyEventOID, studySubjectLabel, formOID, userAccountBean,ordinalValue);
+                responseDTO = sdvUtil.getFormDetailsForSDV(studyOID, formOID, studyEventOID, studySubjectLabel, ordinalValue, changedAfterSdvOnlyFilterFlag);
+            }
+            catch(OpenClinicaSystemException e) {
+                e.printStackTrace();
+                return new ResponseEntity<>(validateService.getResponseForException(e, studyOID, ""), HttpStatus.BAD_REQUEST);
+            }
+            return new ResponseEntity<>(responseDTO, HttpStatus.OK);    
         }
-        catch(OpenClinicaSystemException e) {
+        catch (Exception e) {
+            StringWriter errors = new StringWriter();
+            e.printStackTrace(new PrintWriter(errors));
             e.printStackTrace();
-            return new ResponseEntity<>(validateService.getResponseForException(e, studyOID, ""), HttpStatus.BAD_REQUEST);
+            return new ResponseEntity<>(errors.toString(), HttpStatus.BAD_REQUEST);
```

Comment:
I don't think it's a good idea to return the stacktrace in the API response. If the API is already returning 400, my guess is this is failing even before this method is executed. Have you checked the server logs to see if there are any errors there?

Reply:
Hi Shiva, I don't have access to the server log. How to check it? Let's discuss on Slack.